### PR TITLE
fix: Refactor no default BSL code validation

### PR DESCRIPTION
**Pull request by mateusoliveira43**
## Description

Refactor validation of no default BSL and also check error messages of validations.

## How to test

Run `make test`.

### CODE REVIEW

File: controllers/validator.go [hunk 1]
```diff
@@ -53,8 +52,8 @@ func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error)
 			return false, errors.New("BackupLocation must have cloud storage prefix when backupImages is not set to false")
 		}
 
-		// Check the Velero flags 'no-secret' or 'no-default-backup-location' are not set
-		if !(dpa.Spec.Configuration.Velero.HasFeatureFlag("no-secret") || dpa.Spec.Configuration.Velero.NoDefaultBackupLocation) {
+		// Check if the Velero feature flag 'no-secret' is not set
+		if !dpa.Spec.Configuration.Velero.HasFeatureFlag("no-secret") {
```
Comment (mateusoliveira43) [inline controllers/validator.go:56]: We do not need to check `NoDefaultBackupLocation` here. If code reaches here, it means `NoDefaultBackupLocation` is false

**Review by kaovilai — APPROVED** [review]

**Review by shubham-pampattiwar — APPROVED** [review]

**Comment by kaovilai** [comment]
once https://github.com/openshift/oadp-operator/pull/1294 merges, cherry-pick should work.

**Comment by kaovilai** [comment]
> tide Pending — Not mergeable. Retesting 14 jobs.

it's retesting against latest HEAD from prior merges.. patience :)

**Review by sseago — APPROVED** [review]

**Comment by kaovilai** [comment]
/override ci/prow/4.13-e2e-test-azure
/unhold
from virt backup suite which we discussed should not fail

**Comment by mateusoliveira43** [comment]
/override ci/prow/4.14-e2e-test-aws
snapshots errors and PROW timeout (that we need to investigate, are our E2E tests taking more than 2 hours to run?), and want to unblock Sachin's PR

**Pull request merged**